propagator/xray/test/text_map_extractor_test.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# Copyright OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+describe OpenTelemetry::Propagator::XRay::TextMapExtractor do
+  let(:extractor) { OpenTelemetry::Propagator::XRay::TextMapExtractor.new }
+
+  describe('#extract') do
```

Comment:
I am not familiar with the XRay implementation so apologies if this is not relevant. The Java and .Net implementations have a test case for "additional fields" and if I am reading this right it seems to ignore them:

https://github.com/open-telemetry/opentelemetry-java/blob/f64ec29f92b520378caa756262ba98980cbd747e/extensions/aws/src/test/java/io/opentelemetry/extension/aws/AwsXrayPropagatorTest.java#L169

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/06300ef28d8a8db61dd889c03cab1b3fe7fefed6/test/OpenTelemetry.Contrib.Extensions.AWSXRay.Tests/Trace/TestAWSXRayPropagator.cs#L223

However it looks like the `XRAY_CONTEXT_REGEX` will not match a header that includes additional fields. Is that OK? 

For symmetry I think we should consider having this behave similar to the other implementations or open an issue there if this is the correct behavior. 

```
irb(main):004:0> XRAY_CONTEXT_REGEX = /\ARoot=(?<trace_id>([a-z0-9\-]{35}))(?:;Parent=(?<span_id>([a-z0-9]{16})))?(?:;Sampled=(?<sampling_state>[01d](?![0-9a-f])))?\Z/.freeze
=> /\ARoot=(?<trace_id>([a-z0-9\-]{35}))(?:;Parent=(?<span_id>([a-z0-9]{16})))?(?:;Sampled=(?<sampling_state>[01d](?![0-9a-f])))?\Z/
irb(main):005:0> header = "Root=1-8a3c60f7-d188f8fa79d48a391a778fa6;Parent=53995c3f42cd8ad8;Sampled=1;Foo=Bar"
=> "Root=1-8a3c60f7-d188f8fa79d48a391a778fa6;Parent=53995c3f42cd8ad8;Sampled=1;Foo=Bar"
irb(main):006:0> match = header.match(XRAY_CONTEXT_REGEX)
=> nil
```

Reply:
I've adjusted the REGEX to allow for extra fields and added a new test that demonstrates this in action.

I've piped these new fields into a `Tracestate` structure inside the `SpanContext`. The reason these fields are ignored in other implementations (and why it's ultimately ignored in this one even though the fields are being sent to a `Tracestate` is because we follow https://www.w3.org/TR/trace-context/#key for `Tracestate` which state a key should not start with an uppercase letter... and XRay seems to do all uppercases for extra fields.